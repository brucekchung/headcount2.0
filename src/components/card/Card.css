.card {
  border: 1px solid black;
}

.under-fifty {
  color: red;
}