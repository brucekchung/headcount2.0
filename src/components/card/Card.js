import React from 'react';
import './Card.css';
import { string, object } from 'prop-types';

const Card = ({ className, location, data, handleCompare }) => {
  const renderedData = () => {
    return Object.keys(data).map((year, index) => {
      const percentage = data[year];
      const name = percentage < 0.5 ? "under-fifty" : "over-fifty"

      return ( 
        <li className={ name } key={ index }>
           { year }: { percentage }
        </li> 
      )
    })
  }
 
  return (
    <div className={ className }
         onClick={ handleCompare }>
      <h2>{ location }</h2>
      <ul>
      { renderedData() }
      </ul>
    </div>
  )
}

Card.propTypes = {
  location: string.isRequired,
  data: object.isRequired
}

export default Card