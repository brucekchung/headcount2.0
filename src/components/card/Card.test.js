/* eslint-disable */
import React from 'react'
import Card from './Card'
import {shallow} from 'enzyme'

describe('Card', () => {
  let wrapper
  const data = {
    "Location": "Colorado",
    "TimeFrame": 2007,
    "DataFormat": "Percent",
    "Data": 0.39465
  }
  const key = 245245098

  beforeEach(() => {
    wrapper = shallow(
      <Card data={data} key={key} className='card' location='denver'/>
    )
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})