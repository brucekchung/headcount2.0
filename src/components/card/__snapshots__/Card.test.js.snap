// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "Data": 0.39465,
            "DataFormat": "Percent",
            "Location": "Colorado",
            "TimeFrame": 2007,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <ul>
          <li
                    className="over-fifty"
          >
                    Location
                    : 
                    Colorado
          </li>
          <li
                    className="over-fifty"
          >
                    TimeFrame
                    : 
                    2007
          </li>
          <li
                    className="over-fifty"
          >
                    DataFormat
                    : 
                    Percent
          </li>
          <li
                    className="under-fifty"
          >
                    Data
                    : 
                    0.39465
          </li>
</ul>,
      ],
      "className": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="over-fifty"
>
              Location
              : 
              Colorado
</li>,
            <li
              className="over-fifty"
>
              TimeFrame
              : 
              2007
</li>,
            <li
              className="over-fifty"
>
              DataFormat
              : 
              Percent
</li>,
            <li
              className="under-fifty"
>
              Data
              : 
              0.39465
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Location",
                ": ",
                "Colorado",
              ],
              "className": "over-fifty",
            },
            "ref": null,
            "rendered": Array [
              "Location",
              ": ",
              "Colorado",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "TimeFrame",
                ": ",
                2007,
              ],
              "className": "over-fifty",
            },
            "ref": null,
            "rendered": Array [
              "TimeFrame",
              ": ",
              2007,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "DataFormat",
                ": ",
                "Percent",
              ],
              "className": "over-fifty",
            },
            "ref": null,
            "rendered": Array [
              "DataFormat",
              ": ",
              "Percent",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Data",
                ": ",
                0.39465,
              ],
              "className": "under-fifty",
            },
            "ref": null,
            "rendered": Array [
              "Data",
              ": ",
              0.39465,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <ul>
            <li
                        className="over-fifty"
            >
                        Location
                        : 
                        Colorado
            </li>
            <li
                        className="over-fifty"
            >
                        TimeFrame
                        : 
                        2007
            </li>
            <li
                        className="over-fifty"
            >
                        DataFormat
                        : 
                        Percent
            </li>
            <li
                        className="under-fifty"
            >
                        Data
                        : 
                        0.39465
            </li>
</ul>,
        ],
        "className": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="over-fifty"
>
                Location
                : 
                Colorado
</li>,
              <li
                className="over-fifty"
>
                TimeFrame
                : 
                2007
</li>,
              <li
                className="over-fifty"
>
                DataFormat
                : 
                Percent
</li>,
              <li
                className="under-fifty"
>
                Data
                : 
                0.39465
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Location",
                  ": ",
                  "Colorado",
                ],
                "className": "over-fifty",
              },
              "ref": null,
              "rendered": Array [
                "Location",
                ": ",
                "Colorado",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "TimeFrame",
                  ": ",
                  2007,
                ],
                "className": "over-fifty",
              },
              "ref": null,
              "rendered": Array [
                "TimeFrame",
                ": ",
                2007,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "DataFormat",
                  ": ",
                  "Percent",
                ],
                "className": "over-fifty",
              },
              "ref": null,
              "rendered": Array [
                "DataFormat",
                ": ",
                "Percent",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Data",
                  ": ",
                  0.39465,
                ],
                "className": "under-fifty",
              },
              "ref": null,
              "rendered": Array [
                "Data",
                ": ",
                0.39465,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
